src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/RunningJobs.java

```diff
@@ -233,8 +233,12 @@ private void cancelMatchingJobs(GerritTriggeredEvent event, String jobName, Caus
            List<Queue.Item> itemsInQueue = Queue.getInstance().getItems((Queue.Task)getJob());
            for (Queue.Item item : itemsInQueue) {
                if (checkCausedByGerrit(event, item.getCauses())) {
-                   if (jobName.equals(item.task.getName())) {
+                   // Use transformed fullDisplayNameFor to avoid
+                   // ambigous comparing between same name projects but located at different folders
+                   // FolderName » jobName to FolderName/jobName
+                   if (jobName.equals(item.task.getFullDisplayName().replace(" » ", "/"))) {
```

Comment:
The problem here is that the `displayName` can be anything and change almost at any time and doesn't need to be anything like `jobName` that is gained from `getName`/`getFullName`

Reply:
Another option is change the strict comparison between jobNames to regexp something like ".*jobName$"
But in this case we will have collision with projects located at the different projects folders with same names. This is quite less possible scenario, then I'm going to fix. 
@rsandell what do you think is it ok to fix it this way?